src/environmentd/src/http/sql.rs

```diff
@@ -1279,6 +1278,29 @@ async fn execute_stmt<S: ResultSender>(
             ),
             ctx_extra,
         },
+        ExecuteResponse::Copying { rx, .. } => {
+            let num_rows = match sender.await_rows(client.canceled(), rx).await? {
+                Ok(CopyToResponseUnary::Rows(rows)) => {
+                    rows
+                }
+                Ok(CopyToResponseUnary::Error(e)) => {
+                    return Ok(
+                        SqlResult::err(client, Error::Unstructured(anyhow!(e))).into(),
+                    );
+                }
+                Ok(CopyToResponseUnary::Canceled) => {
+                    return Ok(SqlResult::err(client, AdapterError::Canceled).into());
+                }
+
+                Err(e) => {
+                    return Ok(
+                        SqlResult::err(client, Error::Unstructured(anyhow!(e))).into(),
+                    );
+                },
+            };
+            let tag = format!("COPY {}", num_rows);
+            SqlResult::ok(client, tag, Vec::default()).into()
+        }
```

Comment:
Mentioned this briefly on Slack, but I think you probably _don't_ want to plumb the copy response into the pgwire protocol layer itself. Instead I think you want to aggregate the responses in the adapter layer, in the `ActiveCopyTo` processing, and just plumb through an `ExecuteResponse` to here.

Reply:
I pushed an update, directly sending back an `ExecuteResponse::Copied` instead of this custom thing.